src/Billing.php

```diff
@@ -19,6 +19,36 @@
  */
 class Billing {
 
+	/**
+	 * Initialize Billing actions and Action Scheduler hooks.
+	 *
+	 * @since x.x.x
+	 */
+	public static function schedule_event() {
+		add_action( Heartbeat::DAILY, array( __CLASS__, 'handle_billing_setup_check' ) );
+	}
+
+	/**
+	 * Check if the advertiser has set the billing data.
+	 *
+	 * @since x.x.x
+	 *
+	 * @return mixed
+	 */
+	public function handle_billing_setup_check() {
+
+		Pinterest_For_Woocommerce()::add_billing_setup_info_to_account_data();
+
+		/**
+		 * Redeem credits if billing is setup and credits are not already redeemed.
+		 */
+		if ( Pinterest_For_Woocommerce()::get_billing_setup_info_from_account_data() && ! Pinterest_For_Woocommerce()::get_redeem_credits_info_from_account_data() ) {
```

Comment:
Please use separate handle for this with higher priority number to ensure that it runs after billing setup check. This way we will have cleaner flow and we will be able to modify frequency of both events independently as I think it will be necessary.

Reply:
That makes sense!
I placed priority 20 for the redeem credits action as a placeholder.
The priorities is not totally needed at this point, but as you mentioned is a nice to have for future modifications :ok_hand: